src/Aspire.Hosting.Azure/AzureCosmosDBResource.cs

```diff
@@ -22,5 +22,32 @@ public class AzureCosmosDBResource(string name, string? connectionString)
     /// Gets the connection string to use for this database.
     /// </summary>
     /// <returns>The connection string to use for this database.</returns>
-    public string? GetConnectionString() => ConnectionString;
+    public string? GetConnectionString() => IsEmulator
+        ? AzureCosmosDBEmulatorConnectionString.Create(GetEmulatorPort("emulator"))
+        : ConnectionString;
+
+    /// <summary>
+    /// Gets a value indicating whether the Azure Cosmos DB resource is running in the local emulator.
+    /// </summary>
+    public bool IsEmulator => this.IsContainer();
+
+    private int GetEmulatorPort(string endpointName) =>
+        Annotations
+            .OfType<EndpointAnnotation>()
+            .FirstOrDefault(x => x.Name == endpointName)
+            ?.Port
+        ?? throw new DistributedApplicationException($"Azure Cosmos DB resource does not have endpoint annotation with name '{endpointName}'.");
+}
+
+static file class AzureCosmosDBEmulatorConnectionString
```

Comment:
```suggestion
file static class AzureCosmosDBEmulatorConnectionString
```

The visibility goes first. `public static` not `static public`.

Reply:
Do you want me to change the .editorconfig too then? Since the build is now failing due to this: https://github.com/dotnet/aspire/pull/1244/checks?check_run_id=20240521405

Reply:
See bb65e14261e4ad32d3280c0bdb4f9905723acf33